File: app/src/main/java/za/co/duartbreedt/androidconceptsreference/services/employee/EmployeeService.java
package za.co.duartbreedt.androidconceptsreference.services.employee;

import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.GET;
import retrofit2.http.POST;
import retrofit2.http.PUT;
import retrofit2.http.Path;
import retrofit2.http.Query;
import za.co.duartbreedt.androidconceptsreference.services.employee.domain.Employee;

public interface EmployeeService {

    @GET("users/{id}")
    Call<Employee> getEmployee(@Path("id") int employeeId);

    @GET("users")
    Call<Employee> getEmployees(@Query("tenure") int years);

    @POST("users")
    Call<Employee> createEmployee(@Body Employee employee);

    @PUT("users/{id}")
    Call<Employee> getEmployee(
        @Path("id") int employeeId,
        @Body Employee employee
    );
}

File: app/src/main/java/za/co/duartbreedt/androidconceptsreference/services/employee/EmployeeService.kt
package za.co.duartbreedt.androidconceptsreference.services.employee

import retrofit2.Response
import retrofit2.http.*
import za.co.duartbreedt.androidconceptsreference.services.employee.domain.Employee

interface EmployeeService {

    @GET("users/{id}")
    suspend fun getEmployee(@Path("id") employeeId: Int): Response<Employee>

    @GET("users")
    suspend fun getEmployees(@Query("tenure") years: Int): Response<Employee>

    @POST("users")
    suspend fun createEmployee(@Body employee: Employee): Response<Employee>

    @PUT("users/{id}")
    suspend fun getEmployee(
        @Path("id") employeeId: Int,
        @Body employee: Employee
    ): Response<Employee>
}